esptool/loader.py

```diff
@@ -243,7 +243,10 @@ def __init__(self, port=DEFAULT_PORT, baud=ESP_ROM_BAUD, trace_enabled=False):
         self.stub_is_disabled = False
 
         if isinstance(port, str):
-            self._port = serial.serial_for_url(port)
+            try:
+                self._port = serial.serial_for_url(port)
+            except serial.serialutil.SerialException:
+                raise FatalError("No device detected on the given port.")
```

Comment:
Suggestion: Mention the port here. So when the user copy-pastes this error somewhere to get help, people can ask about the particular port. Something like:

```suggestion
                raise FatalError(f"No device detected on port {port}")
```